File: src/TheWorld/wwwroot/js/tripsController.js
//tripsController.js

(function() {

	"use strict";

	//Getting the existin module
	angular.module("app-trips")
		.controller("tripsController", tripsController);

	function tripsController($http) {

		var vm = this;

		vm.trips = [];

		vm.newTrip = {};

		vm.errorMessage = "";
		vm.isBusy = true;

		vm.addTrip = function () {

	   	        vm.isBusy = true;
	            vm.errorMessage = "";

	        $http.post("/api/trips", vm.newTrip)
	            .then(function(response) {
	                    //success
	            	vm.trips.push(response.data);
	                    vm.newTrip = {};
	                },
	                function(error) {
	                    //error
	                    vm.errorMessage = "Failed to save new trip" + error;
	                })
	            .finally(function() {
	                    vm.isBusy = false;
	                }
	            );
	    };

		$http.get("/api/trips")
			.then(function(response) {
				//on success
					angular.copy(response.data, vm.trips);
				},
				function(error) {
					//on failure
				    vm.errorMessage = "Failed to load data" + error;
				})
			.finally(function() {
		        vm.isBusy = false;
		    });

	}

})();

File: src/TheWorld/Controllers/Api/TripsController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using AutoMapper;
using Microsoft.AspNetCore.Authorization;
using Microsoft.AspNetCore.Mvc;
using Microsoft.Extensions.Logging;
using TheWorld.Models;
using TheWorld.ViewModel;

namespace TheWorld.Controllers.Api
{
    [Route("api/trips")]
    [Authorize]
    public class TripsController : Controller

    {

        private IWorldRepository _repository;
        private ILogger _logger;

        public TripsController(IWorldRepository repository, ILogger<TripsController> logger)
        {
            _repository = repository;
            _logger = logger;
        }

        [HttpGet("")]
        public JsonResult Get()
        {
            
                var trips = _repository.GetUserTripsWithStops(User.Identity.Name);

                var results  = Mapper.Map<IEnumerable<TripViewModel>>(trips);
                return Json(results);
            
        }

        [HttpPost("")]
        public async Task<IActionResult> Post([FromBody]TripViewModel theTrip)
        {

            if(ModelState.IsValid)
            { 
                var newTrip = Mapper.Map<Trip>(theTrip);

                newTrip.Username = User.Identity.Name;

                _repository.AddTrip(newTrip);

                if (await _repository.SaveChangesAsync())
                {
                    return Created($"api/trips/{theTrip.Name}", Mapper.Map<TripViewModel>(newTrip));
                }

            
            }

            return BadRequest("Failed To Save Changes To The Database");
        }

    }
}
